Libraries/Azure.psm1

```diff
@@ -1909,6 +1925,16 @@ Function Generate-AzureDeployJSONFile ($RGName, $ImageName, $osVHD, $RGXMLData,
         $vmAdded = $true
         $role = $role + 1
         $vmCount = $role
+        $heteroImageIndex = $heteroImageIndex + 1
```

Comment:
If ARMImageName is single image, there's no value here

Reply:
Surrounded logic with a check for $heteroImages.
        if( $heteroImages )
        {
           ...
        }